src/rgw/rgw_op.cc

```diff
@@ -631,12 +689,11 @@ int rgw_build_bucket_policies(const DoutPrefixProvider *dpp, rgw::sal::Driver* d
     try {
       ret = s->user->read_attrs(dpp, y);
       if (ret == 0) {
-	auto user_policies = get_iam_user_policy_from_attr(s->cct,
-							   s->user->get_attrs(),
-							   s->user->get_tenant());
-          s->iam_user_policies.insert(s->iam_user_policies.end(),
-                                      std::make_move_iterator(user_policies.begin()),
-                                      std::make_move_iterator(user_policies.end()));
+        // load all user and group policies
+        load_iam_identity_policies(dpp, y, driver,
+                                   s->user->get_info(),
+                                   s->user->get_attrs(),
+                                   s->iam_identity_policies);
```

Comment:
i'm seeing RGWPostObj auth failures with account users, because the authentication happens _after_ this call to `rgw_build_bucket_policies()`. so `s->user` at this point is the 'anonymous user', and we're unable to apply any of these identity policies

role policy gets added by `RoleApplier::modify_request_state()` so doesn't have this issue. i think the other Appliers should handle user policy the same way

Reply:
Had a quick test of HEAD 1d033408c07388155c517eec957c6e591538e513. Things look good so far. I'll let our product team to resume their testing on this version.

Reply:
> i'm seeing RGWPostObj auth failures with account users

fixed with commit "rgw/auth: pass user policies into identities"